js/main.js

```diff
@@ -75,6 +103,29 @@ function viewEntries(event) {
   data.view = 'entries';
 }
 
+function editEntry(event) {
+  if (event.target.tagName !== 'I') {
+    return;
+  }
+  viewEntryForm();
+  var entryListElement = event.target.closest('li');
+  data.editing = entryListElement;
+  var entryObject = getEntryObject(entryListElement);
+
+  inputTitle.value = entryObject.title;
+  inputPhotoUrl.value = entryObject.photoUrl;
+  placeholderImage.setAttribute('src', entryObject.photoUrl);
+  inputNotes.value = entryObject.notes;
+}
+
+function getEntryObject(entryListElement) {
+  var entryId = entryListElement.getAttribute('data-entry-id');
+  var numEntries = data.entries.length;
+  var entryIndex = numEntries - entryId;
+  var entryObject = data.entries[entryIndex];
```

Comment:
You should not depend on an `id` to be the same as its index like this. For instance, what if eventually the item with the id `4` gets deleted? Now this won't work. Instead, you should loop over your list of entries and find the entry with the matching id.

Reply:
Ahh yes I forgot about the "Delete an entry" feature of this project 😅 Thank you for the helpful pointers!